Sources/StreamChat/WebSocketClient/EventMiddlewares/MemberEventMiddleware.swift

```diff
@@ -28,33 +40,80 @@ struct MemberEventMiddleware: EventMiddleware {
                 
                 // We remove the member from the channel
                 channel.members.remove(member)
+                if let membership = channel.membership, membership.user.id == event.user.id {
+                    channel.membership = nil
+                }
                 
                 // If there are any MemberListQueries observing this channel,
                 // we need to update them too
                 member.queries.removeAll()
                 
             case let event as NotificationAddedToChannelEventDTO:
-                _ = try session.saveChannel(payload: event.channel, query: nil)
+                let channel = try session.saveChannel(payload: event.channel, query: nil)
+                let member = try session.saveMember(payload: event.member, channelId: event.channel.cid)
+                channel.membership = member
+                
+                // If there are any `MemberListQuery`s observing this Channel
+                // without any filters (so the query observes all members)
+                // the new Member should be linked to them too
+                // so `MemberListController` works as expected
+                // To make it work with queries with filters, we need to mirror `ChannelListController` logic
+                // `shouldListUpdatedChannel` and such
+                channel.memberListQueries.filter { $0.filterJSONData == nil }.forEach {
+                    $0.members.insert(member)
+                }
```

Comment:
Would it be possible to encapsulate this in some private func to avoid code duplication? 🙏